girder/models/setting.py

```diff
@@ -171,6 +171,12 @@ def validateCoreBrandName(doc):
             raise ValidationException('The brand name may not be empty', 'value')
 
     @staticmethod
+    @setting_utilities.validator(SettingKey.BANNER_COLOR)
+    def validateCoreBannerColor(doc):
+        if not doc['value']:
```

Comment:
We can probably do a better validation than just this. A regex would work well here.

Reply:
Do you have an idea why this kind of verification doesn't work ? 

![screenshot from 2017-09-11 11-01-44](https://user-images.githubusercontent.com/29961430/30281510-a0f771f4-96e0-11e7-9920-4a9c6b6c6cb3.png)

Reply:
You're throwing an error if the regex matches.  Try inverting the condition.

Reply:
That regex should work for hex colors. Is there a reason we want to limit this to just hex? I could see an admin wanting to just type one of the named colors like "white".

Reply:
Oh right ! Yes it's true, I will change that

Reply:
@zachmullen When this widget is clicked, a color picker appears, which automatically fills in in hex values, and does not seem to allow the manual entry of non-hex color values. While it's technically possible to submit a color alias via an API call, I think it would simplify a lot of assumptions (maybe with security implications) if we just require hex colors to be used here.

Reply:
![screenshot from 2017-09-11 13-22-05](https://user-images.githubusercontent.com/29961430/30287764-481b7954-96f4-11e7-9235-be7a573faffe.png)

If we enter a string like "blue", "red", etc.. The color change and the string is converted to an hexadecimal value (Mozilla color picker)
So it doesn't raise an error even if we only check for hexadecimal value
Moreover I think the check of different color name will be such a pain and not very fundamental